mlir-sycl/lib/Dialect/SYCL/IR/SYCLOps.cpp

```diff
@@ -44,6 +44,13 @@ static LogicalResult checkSameDimensions(Operation *op, Type thisType,
                    << inputDims << " vs " << thisDims;
 }
 
+template <typename Ty> static Ty getMemRefElementType(Type ty) {
+  auto mt = dyn_cast<MemRefType>(ty);
+  if (!mt)
+    return {};
+  return dyn_cast<Ty>(mt.getElementType());
```

Comment:
Nit / matter of taste:
```suggestion
return mt ? dyn_cast<Ty>(mt.getElementType()) : {};
```

Reply:
This was part of a different PR. Now that I rebased it's gone